### PR TITLE
Simplify the storage reset

### PR DESCRIPTION
**Pull request by poncovka**
Simplify the code that handles the storage reset.

### CODE REVIEW

File: pyanaconda/storage/initialization.py [hunk 1]
```diff
@@ -137,6 +138,11 @@ def reset_storage(storage, teardown=False, retry=True):
         except Exception:  # pylint: disable=broad-except
             log_exception_info(log.error, "Failure tearing down device tree.")
 
+    # Clear the exclusive disks to scan all devices in the system.
+    if scan_all:
+        disk_select_proxy = STORAGE.get_proxy(DISK_SELECTION)
+        disk_select_proxy.SetExclusiveDisks([])
```
Comment (jkonecny12) [inline pyanaconda/storage/initialization.py]: I would be cautious about this. In general any device which is not supported by blivet may crash the system.
If I'm not mistaken all the disks in the protected and excluded list should be scanned by blivet or we will crash in some situations.

Comment (poncovka) [reply pyanaconda/storage/initialization.py]: This only replaces the workaround with selected disks. We already scan all devices if the storage configuration fails and if a user requests to rescan a storage in TUI.

The only place I have changed is the logic for the Rescan button in GUI to make it consistent with TUI. However, I am ready to revert this change if it causes problems.

**Comment by poncovka** [comment]
Rebased.

File: pyanaconda/storage/initialization.py (hunk 1 above)
Comment (jkonecny12) [reply pyanaconda/storage/initialization.py]: I won't block merging of this but if I'm not mistaken for example if you have broken VG group (missing disk) the blivet library will fail on scan. Now this behavior can be work-around by using the `ignoredisk` command in kickstart. If I'm not mistaken then this won't work now for this part of the code.

You can merge this now but please test how it behaves on broken VG (the easiest test I know) with ignoreddisks command. If didn't introduce a regression by this and if yes then please make a follow-up PR.

**Comment by poncovka** [comment]
Fixed the function `_reset_storage`.

File: pyanaconda/storage/initialization.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/storage/initialization.py]: This pull request should not affect the logic of the `ignoredisk` command at all. Merging.

**Pull request merged**